apps/cloud-player/app.js

```diff
@@ -1,150 +1,81 @@
 var Application = require('@yodaos/application').Application
-var AudioFocus = require('@yodaos/application').AudioFocus
-var speechSynthesis = require('@yodaos/speech-synthesis').speechSynthesis
-var MediaPlayer = require('@yoda/multimedia').MediaPlayer
 var flora = require('@yoda/flora')
 var _ = require('@yoda/util')._
 var logger = require('logger')('cloud-player')
 
-var FloraUri = 'unix:/var/run/flora.sock#cloud-player'
-var GetStreamChannel = 'yodaos.apps.cloud-player.get-stream-channel'
-var MultimediaStatusChannel = 'yodaos.apps.multimedia.playback-status'
-var TtsStatusChannel = 'yodaos.apps.cloud-player.tts.status'
-
-var StatusCode = {
-  start: 0,
-  end: 1,
-  cancel: 2,
-  error: 3
-}
+var constant = require('./constant')
+var FloraUri = constant.FloraUri
 
 var app = Application({
   created: function created () {
+    this.voices = []
     this.agent = new flora.Agent(FloraUri)
     this.agent.start()
   },
   url: function url (urlObj) {
     logger.info('received url', require('url').format(urlObj))
```

Comment:
```suggestion
    logger.info('received url', url.href)
```
Does this work?

Reply:
`href` might not be the up to date full url since we didn't re-format the urlObj if there are any new queries attached to the url. It might be performance burden since most of the cases `href` is not used.

Reply:
But the `url.format()` takes a little bit performance hurt, why not only printing this object?

Reply:
Formatted url has much cleaner outputs (single lined and pretty regular format string). I'd suggest keep this line as it is _for now_.